### PR TITLE
Modification via a type alias failing on Scala 3

### PR DESCRIPTION
**Pull request by OndrejSpanel**
Another Scala 3 regression. Following code fails to compile:

```scala

  case class State(x: Int)

  type S = State

  val s: S = State(0)
  val modified = s.modify(_.x).setTo(1)

  modified.x shouldBe 1
```

The error is:

> Unsupported source object: must be a case class or sealed trait, but got: type S

I think hopefully I should be able to prepare a fix.

### CODE REVIEW

**Comment by KacperFKorban** [comment]
It most likely just needs a `dealias` call after `tpe` at line 163.

**Comment by OndrejSpanel** [comment]
I have added a commit inverting the case class / sealed trait logic. Anything which does not require a special handling is handled as a case class, hoping it will behave like one.

This does not fix the issue yet, but it could make the code a bit more robust when handing unexpected situations.

> it actually requires a bunch of .widen.dealias calls

I will try to drop some at random source location and see where it works. 😊

**Comment by KacperFKorban** [comment]
It's pretty much before any `.typeSymbol` call

**Comment by KacperFKorban** [comment]
> I have added a commit inverting the case class / sealed trait logic. Anything which does not require a special handling is handled as a case class, hoping it will behave like one.

I'm not sure if that's what we want. The condition for `Case` was in place because we need a `copy` to be able to rewrite normal field accesses (not to mention the default methods).
Also, the error message was quite clear. In this case, it was a bug (because of an obvious error of mine) but other times it might be helpful.

**Comment by OndrejSpanel** [comment]
> Also, the error message was quite clear. In this case, it was a bug (because of an obvious error of mine) but other times it might be helpful.

OK. I will leave that out. It is certainly not needed for the fix and it is not related, therefore I should not have mixed it here.

**Comment by OndrejSpanel** [comment]
Tests pass. I think the PR should be ready for review / merge.

File: quicklens/src/test/scala/com/softwaremill/quicklens/ModifyAliasTest.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.scalatest.matchers.should.Matchers
+
+import ModifyAliasTest._
+
+object ModifyAliasTest {
+
+  case class State(x: Int)
+
+  type S = State
+}
```
Comment (KacperFKorban) [inline quicklens/src/test/scala/com/softwaremill/quicklens/ModifyAliasTest.scala:13]:
```suggestion

  sealed trait Expr {
    def i: Int
  }
  case class ListInt(i: Int) extends Expr

  type E = Expr
}
```

**Review by KacperFKorban — CHANGES_REQUESTED** [review]
I added a suggestion with a test case for the other case (`sealed tarit`/`enum`) that would still fail.
I cannot add comments there, but two more `.widen.dealias` calls will be needed before accessing `.typeSymbol.children`.

**Review by KacperFKorban — APPROVED** [review]

**Pull request merged**